model/common.go

```diff
@@ -16,3 +11,13 @@ type AvailabilityStatus struct {
 	LastCheckedAt      time.Time `json:"last_checked_at,omitempty"`
 	LastAvailableAt    time.Time `json:"last_available_at,omitempty"`
 }
+
+type PauseResponse struct {
+	PausedAt *string `json:"paused_at,omitempty"`
+}
+
+type AvailabilityStatusResponse struct {
+	AvailabilityStatus *string `json:"availability_status,omitempty"`
+	LastCheckedAt      *string `json:"last_checked_at,omitempty"`
+	LastAvailableAt    *string `json:"last_available_at,omitempty"`
+}
```

Comment:
I agree with this, if we made them all pointers then they wouldn't get stored in the database when they aren't there which might solve all of our problems.

Reply:
You should be able to make `util.DateTimeToRFC3339()` accept an interface as a param instead of a `time.Time` struct, and do the right thing based on the type. Just do a nil check and return nil if the time is nil.

Code Example here: https://go.dev/play/p/kP_WRj6tkgj